### PR TITLE
Enhance test_verify_fec_histogram logic to handle a stable link with stale transient FEC symbol errors

### PR DESCRIPTION
**Pull request by kewei-arista**
### Description of PR

`platform_tests/test_intf_fec.py` `test_verify_fec_histogram` checks no critical FEC bins for a link and may fail the test if there are some errors with the link. However, this test doesn't consider the transient symbol errors accumulated during interface state transition (or other transient state) and thus it can also fail for a stable link with stale errors during the testing time.

For example, it may fail with following BIN output:
```
(Pdb) intf_name
'Ethernet280'
(Pdb) fec_hist
[{'symbol errors per codeword': 'BIN0', 'codewords': '99235826365'},
{'symbol errors per codeword': 'BIN1', 'codewords': '406154'}, {'symbol
errors per codeword': 'BIN2', 'codewords': '1781'}, {'symbol errors per
codeword': 'BIN3', 'codewords': '0'}, {'symbol errors per codeword':
'BIN4', 'codewords': '0'}, {'symbol errors per codeword': 'BIN5',
'codewords': '0'}, {'symbol errors per codeword': 'BIN6', 'codewords':
'0'}, {'symbol errors per codeword': 'BIN7', 'codewords': '0'}, {'symbol
errors per codeword': 'BIN8', 'codewords': '0'}, {'symbol errors per
codeword': 'BIN9', 'codewords': '0'}, {'symbol errors per codeword':
'BIN10', 'codewords': '0'}, {'symbol errors per codeword': 'BIN11',
'codewords': '0'}, {'symbol errors per codeword': 'BIN12', 'codewords':
'0'}, {'symbol errors per codeword': 'BIN13', 'codewords': '0'},
{'symbol errors per codeword': 'BIN14', 'codewords': '0'}, {'symbol
errors per codeword': 'BIN15', 'codewords': '1'}]
```

Note that there's only 1 BIN15 symbol error and it's not incrementing for a long time, so this link is stable. However, the test can still fail in this case.

This change is trying to enhance the test case to handle these stale errors by checking whether a test interface is susceptible to this issue with the 1st snapshot of fec histogram. If so, it will extend waiting time to 10 minutes for each loop and that's 20 minutes in total and make sure no critical BINs will ever increment during this 20 min period. The rational is that it's very likely to see the changes in these BINs in this long window if the link is not stable. Usually we can see the changes in every seconds for a marginal link.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [x] Test case improvement

### Back port request
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [x] 202505
- [x] 202511
- [x] msft-202412
- [x] msft-202503

#### What is the motivation for this PR?
Improve the pass rate by handling these corner cases

#### How did you do it?
Wait for enough long time to make sure the links are actually stable

#### How did you verify/test it?
Confirmed the test now can pass with a stable link but transient FEC symbol errors

### CODE REVIEW

**Review by prgeor — APPROVED** [review]

File: tests/platform_tests/test_intf_fec.py [hunk 1]
```diff
@@ -260,8 +272,26 @@ def test_verify_fec_histogram(duthosts, enum_rand_one_per_hwsku_frontend_hostnam
...
-        for _ in range(3):
-            if not validate_fec_histogram(duthost, intf_name):
+        valid, fec_hist = validate_fec_histogram(duthost, intf_name, True)
+        if not valid:
+            logging.info("Update test sleep time to 10 min due to bin errors in the initial snapshot")
+            sleep_time = 10 * 60
+        snapshots[intf_name] = fec_hist
+
+    for _ in range(2):
+        time.sleep(sleep_time)
```
Comment (prgeor) [inline tests/platform_tests/test_intf_fec.py:291]: @kewei-arista I am not sure this 10min/20min wait time would be acceptable for all platform owners. Can we make this test attribute driven?

**Comment by r12f** [comment]
hi @kewei-arista , do you mind to help bring this to 202412?

**Comment by kewei-arista** [comment]
> hi @kewei-arista , do you mind to help bring this to 202412?

@r12f I created this PR to backport this patch to 202412: https://github.com/Azure/sonic-mgmt.msft/pull/1014

File: tests/platform_tests/test_intf_fec.py (hunk 1 above)
Comment (kewei-arista) [reply tests/platform_tests/test_intf_fec.py:291]: This has been addressed in https://github.com/sonic-net/sonic-mgmt/pull/22733

**Comment by StormLiangMS** [comment]
@kewei-arista could you file separate PR to conflict branches?
